MathLib/InterpolationAlgorithms/PiecewiseLinearInterpolation.cpp

```diff
@@ -48,17 +68,59 @@ double PiecewiseLinearInterpolation::getValue(double pnt_to_interpolate) const
         return _values_at_supp_pnts[_supp_pnts.size() - 1];
     }
 
+    // search interval that has the point inside
     auto const& it(std::lower_bound(_supp_pnts.begin(), _supp_pnts.end(),
                                     pnt_to_interpolate));
     std::size_t const interval_idx = std::distance(_supp_pnts.begin(), it) - 1;
 
-    // compute linear interpolation polynom: y = m * (x - support[i]) + value[i]
-    const double m((_values_at_supp_pnts[interval_idx + 1] -
-                    _values_at_supp_pnts[interval_idx]) /
-                   (_supp_pnts[interval_idx + 1] - _supp_pnts[interval_idx]));
+    return interpolate(_supp_pnts, _values_at_supp_pnts, interval_idx,
+                       pnt_to_interpolate);
+}
 
-    return m * (pnt_to_interpolate - _supp_pnts[interval_idx]) +
-           _values_at_supp_pnts[interval_idx];
+double PiecewiseLinearInterpolation::getInverseValue(double const value) const
+{
+    std::size_t interval_idx = 0;
```

Comment:
On demand, the monotonic is only checked once immediately after an instance the class is created.